routingpy/valhalla_attributes.py

```diff
@@ -0,0 +1,421 @@
+        an edge. But since an edge in Valhalla can possibly go up and down over several hills it
+        might not equate to what most folks think of as grade.
+        """
+        return self._weighted_grade
+
+    @property
+    def road_class(self) -> Optional[RoadClass]:
+        """
+        Returns the :class:`RoadClass` of this edge.
+        """
+        return self._road_class
+
+    @property
+    def speed(self) -> Optional[int]:
+        """
+        Returns the actual speed of the edge, as used by Valhalla.
+        """
+        return self._speed
+
+    @property
+    def length(self) -> Optional[int]:
+        """
+        The length of this edge in meters.
+        """
+        return self._length
+
+    def __repr__(self):  # pragma: no cover
+        return "Edge({})".format(", ".join([f"{k[1:]}: {v}" for k, v in vars(self).items() if v]))
+
+
+class MatchedPoint:
+    """
+    A single matched point
+    """
+
+    def __init__(self, point: dict):
+        self._geometry: List[float] = [point["lon"], point["lat"]]
+        self._match_type = MatchType(point.get("type", "")) or None
+        self._dist_along_edge: Optional[float] = point.get("distance_along_edge")
+        self._dist_from_input: Optional[int] = point.get("distance_from_trace_point")
+        self._edge_index: Optional[int] = point.get("distance_along_edge")
+        self._discontinuity: Optional[MatchDiscontinuity] = None
+        if point.get("begin_route_discontinuity"):
+            self._discontinuity = MatchDiscontinuity("begin")
+        elif point.get("begin_route_discontinuity"):
+            self._discontinuity = MatchDiscontinuity("end")
+
+    @property
+    def geometry(self) -> Union[Tuple[float, float], List[float]]:
+        """
+        The geometry of the point as [lon1, lat1] list.
+        """
+        return self._geometry
+
+    @property
+    def match_type(self) -> MatchType:
+        """
+        Returns the match type.
+        """
+        return self._match_type
+
+    @property
+    def distance_along_edge(self) -> Optional[float]:
+        """
+        Returns the relative distance along the matched edge. E.g. if the point
+        projects to the center of an edge, the return value will be 0.5.
+        """
+        return self._dist_along_edge
+
+    @property
+    def distance_from_input(self) -> Optional[int]:
+        """
+        Returns the distance of the matched point from the corresponding input location.
+        """
+        return self._dist_from_input
+
+    @property
+    def edge_index(self) -> Optional[int]:
+        """
+        Returns the edge index.
+        """
+        return self._edge_index
+
+    @property
+    def discontinuity(self) -> Optional[MatchDiscontinuity]:
+        """
+        Returns the :class:`MatchDiscontinuity` status.
+        """
+        return self._discontinuity
+
+
+class MatchedResults:
```

Comment:
this is what the user will get. he can then go through the `matched_edges` and `matched_points` lists